stop_and_wait_plugin/include/stop_and_wait_plugin.hpp

```diff
@@ -104,9 +104,17 @@ class StopandWait
    */
   void splitPointSpeedPairs(const std::vector<PointSpeedPair>& points, std::vector<lanelet::BasicPoint2d>* basic_points,
                             std::vector<double>* speeds) const;
-
+  /**
+   * \brief Combines the points, times, and yaws to create a trajectory that starts from startTime
+   *  NOTE: If times has non-normal value (nan, inf etc), which can happen when speed is 0, it increments current time by 10hr from previous
+   * \param points The list of points that define the trajectory
+   * \param times The list if times associated to points. The input is modified if times has inf or nan values
+   * \param yaws The list if yaws associated to points
+   * \param starTime start time of the trajectory
+   * \return A trajectory plan
+   */
   std::vector<carma_planning_msgs::msg::TrajectoryPlanPoint> trajectory_from_points_times_orientations(
-      const std::vector<lanelet::BasicPoint2d>& points, const std::vector<double>& times,
+      const std::vector<lanelet::BasicPoint2d>& points, std::vector<double>& times,
```

Comment:
Are we modifying the times passed here? Why did we remove the const keyword

Reply:
yeah the logic is modifying the `times` solely to print it later outside the function for debugging purpose. It is much more meaningful to see the `times` with `downtracks` in the debug message outside, so I purposefully letting it pass by reference. I am also open to other suggestions.